src/edu/csus/ecs/pc2/core/report/FileComparisonUtilities.java

```diff
@@ -0,0 +1,500 @@
+// Copyright (C) 1989-2023 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.core.report;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import edu.csus.ecs.pc2.core.Constants;
+import edu.csus.ecs.pc2.core.StringUtilities;
+import edu.csus.ecs.pc2.core.Utilities;
+import edu.csus.ecs.pc2.core.execute.ExecuteUtilities;
+import edu.csus.ecs.pc2.core.imports.clics.CLICSAward;
+import edu.csus.ecs.pc2.core.imports.clics.CLICSScoreboard;
+import edu.csus.ecs.pc2.core.imports.clics.FieldCompareRecord;
+import edu.csus.ecs.pc2.core.imports.clics.FileComparison;
+import edu.csus.ecs.pc2.core.log.StaticLog;
+import edu.csus.ecs.pc2.core.standings.json.TeamScoreRow;
+
+/**
+ * Utilities for file comparison
+ * 
+ * @author Douglas A. Lane, PC^2 team pc2@ecs.csus.edu
+ */
+public class FileComparisonUtilities {
+    
+    private static  ObjectMapper objectMapper = null;
+
+    public static FileComparison createScoreboardJSONFileComparison(String tsvFileName, String sourceDir, String targetDir, IFileComparisonKey fileComparisonKey) {
+
+        String firstFilename = sourceDir + File.separator + tsvFileName;
+        String secondFilename = targetDir + File.separator + tsvFileName;
+        FileComparison fileComparison = new FileComparison(firstFilename, secondFilename);
+
+        long numberRows = 0;
+
+        try {
+            
+            List<TeamScoreRow> firstTeamScoreRows = FileComparisonUtilities.loadTeamRows(fileComparison.getFirstFilename());
+            List<TeamScoreRow> secondTeamScoreRows = FileComparisonUtilities.loadTeamRows(fileComparison.getSecondFilename());
+
+            int minLines = firstTeamScoreRows.size();
+            if (secondTeamScoreRows.size() < minLines) {
+                minLines = secondTeamScoreRows.size();
+            }
+            
+            numberRows += minLines;
+
+            for (int rowNum = 0; rowNum < minLines; rowNum++) {
+                TeamScoreRow firstRow = firstTeamScoreRows.get(rowNum);
+                TeamScoreRow teamScoreRow = firstTeamScoreRows.get(rowNum);
```

Comment:
I do not understand the above two lines of code.  They are both fetching the exact same value (`firstTeamScoreRows.get(rowNum)`), and assigning that (same) value to two different variables (`firstRow` and `teamScoreRow`).  Subsequently, the following lines of code construct comparisons between those (identical) values, e.g.:
```
                String valueOne = Integer.toString(firstRow.getRank());
                String valueTwo = Integer.toString(teamScoreRow.getRank());
                FieldCompareRecord fieldCompareRecord = new FieldCompareRecord(fieldName, valueOne, valueTwo, null, valueOne);
```
This appears to mean that the "compare records" will _always_ be comparing a value with itself -- which would seem to be a bug.

I'm marking this as a SHOW STOPPER; it either needs to be fixed (if it really is a bug) or I would like an explanation of why it is constructing comparisons of identical values.